rasa/shared/utils/common.py

```diff
@@ -13,24 +17,45 @@ def class_from_module_path(
 ) -> Any:
     """Given the module name and path of a class, tries to retrieve the class.
 
-    The loaded class can be used to instantiate new objects."""
-    # load the module, will raise ImportError if module cannot be loaded
+    The loaded class can be used to instantiate new objects.
+
+    Args:
+        module_path: either an absolute path to a Python class,
+                     or the name of the class in the local / global scope.
+        lookup_path: a path where to load the class from, if it cannot
+                     be found in the local / global scope.
+
+    Returns:
+        a Python class
+
+    Raises:
+        ImportError, in case the Python class cannot be found.
```

Comment:
technically also a `TypeError` in case `import_module` fails

Reply:
not sure I follow 🤔 doesn't it raise a `ModuleNotFoundError` which inherits `ImportError`?

Reply:
`import_module` in 3.8.6 also raises a `TypeError` which inherits from `Exception`

```python
def import_module(name, package=None):
    """Import a module.

    The 'package' argument is required when performing a relative import. It
    specifies the package to use as the anchor point from which to resolve the
    relative import to an absolute import.

    """
    level = 0
    if name.startswith('.'):
        if not package:
            msg = ("the 'package' argument is required to perform a relative "
                   "import for {!r}")
            raise TypeError(msg.format(name))
        for character in name:
            if character != '.':
                break
            level += 1
    return _bootstrap._gcd_import(name[level:], package, level)
```

Reply:
🤔 yeah ok. It sounds a bit more like a detail. Fine if I keep it like it is?

Reply:
It would be more transparent for the caller as they'd know which errors to handle without having to dive deeper into the code, but up to you as it's a detail as you said.